core/chaincode/shim/chaincode.go

```diff
@@ -572,11 +572,11 @@ func (stub *ChaincodeStub) GetRows(tableName string, key []Column) (<-chan Row,
 	if err != nil {
 		return nil, fmt.Errorf("Error fetching rows: %s", err)
 	}
-	defer iter.Close()
 
 	rows := make(chan Row)
 
 	go func() {
+		defer iter.Close()
 		for iter.HasNext() {
 			_, rowBytes, err := iter.Next()
 			if err != nil {
```

Comment:
That change LGTM.

However, should we also exit the loop when err != nil instead of calling close(..) and continuing ? If the channel is closed any point in adding other rows to it ? Also, if there *is* an error, don't we want to provide that to the user as well ? 

But defer to @srderson